src/plugins/discover/public/application/main/components/layout/use_discover_histogram.ts

```diff
@@ -50,233 +41,214 @@ export const useDiscoverHistogram = ({
-
-  useEffect(() => {
-    if (!timeField) {
-      return;
-    }
-    getVisualizeInformation(
-      getUiActions(),
-      timeField,
-      dataView,
-      savedSearch.columns || [],
-      []
-    ).then((info) => {
-      setCanVisualize(Boolean(info));
-    });
-  }, [dataView, savedSearch.columns, timeField]);
+  const [unifiedHistogram, setUnifiedHistogram] = useState<UnifiedHistogramApi>();
 
-  const onEditVisualization = useCallback(
-    (lensAttributes: TypedLensByValueInput['attributes']) => {
-      if (!timeField) {
+  const initializeUnifiedHistogram = useCallback(
+    (api?: UnifiedHistogramApi) => {
+      if (!api || api.initialized || !isInitialized) {
         return;
       }
-      lens.navigateToPrefilledEditor({
-        id: '',
-        timeRange: data.query.timefilter.timefilter.getTime(),
-        attributes: lensAttributes,
-      });
-    },
-    [data.query.timefilter.timefilter, lens, timeField]
-  );
 
-  /**
-   * Height
-   */
+      const {
+        hideChart: chartHidden,
+        interval: timeInterval,
+        breakdownField,
+      } = stateContainer.appState.getState();
 
-  const [topPanelHeight, setTopPanelHeight] = useState(() => {
-    const storedHeight = storage.get(HISTOGRAM_HEIGHT_KEY);
-    return storedHeight ? Number(storedHeight) : undefined;
-  });
+      const { fetchStatus: totalHitsStatus, result: totalHitsResult } =
+        savedSearchData$.totalHits$.getValue();
 
-  const onTopPanelHeightChange = useCallback(
-    (newTopPanelHeight: number | undefined) => {
-      storage.set(HISTOGRAM_HEIGHT_KEY, newTopPanelHeight);
-      setTopPanelHeight(newTopPanelHeight);
-    },
-    [storage]
-  );
+      const { query, filters, time: timeRange } = services.data.query.getState();
 
-  /**
-   * Time interval
-   */
-
-  const onTimeIntervalChange = useCallback(
-    (newInterval: string) => {
-      stateContainer.setAppState({ interval: newInterval });
+      api.initialize({
+        services: { ...services, uiActions: getUiActions() },
+        localStorageKeyPrefix: 'discover',
+        disableAutoFetching: true,
+        initialState: {
+          dataView,
+          query,
+          filters,
+          timeRange,
+          chartHidden,
+          timeInterval,
+          breakdownField,
+          searchSessionId,
+          totalHitsStatus: totalHitsStatus.toString() as UnifiedHistogramFetchStatus,
+          totalHitsResult,
+          requestAdapter: inspectorAdapters.requests,
+        },
+      });
     },
-    [stateContainer]
+    [
+      dataView,
+      inspectorAdapters.requests,
+      isInitialized,
+      savedSearchData$.totalHits$,
+      searchSessionId,
+      services,
+      stateContainer.appState,
+    ]
   );
 
-  /**
-   * Total hits
-   */
-
-  const [localHitsContext, setLocalHitsContext] = useState<UnifiedHistogramHitsContext>();
+  const setUnifiedHistogramApi = useCallback((api: UnifiedHistogramApi | null) => {
```

Comment:
It would be nice to simplify how this happens although that might be dependent upon other lifecycle bits.

Reply:
I greatly simplified this hook overall by storing only the initialized API as a state variable, which avoids additional checks for initialization, and pulling the `isInitialized` logic relying on `searchSessionId` and `isPlainRecord` to `discover_histogram_layout`, which eliminates a lot of the previous complexity.